common/components/componentsBySection/CreateEvent/EventProjectSelectionForm.jsx

```diff
@@ -0,0 +1,145 @@
+  readyForSubmit: () => () => boolean,
+|} & FormPropsBase<FormFields>;
+
+type State = {|
+  selectedProjects: Array<Project>,
+  formIsValid: boolean,
+  validations: $ReadOnlyArray<Validator>,
+|} & FormStateBase<FormFields>;
+
+/**
+ * Encapsulates form for Project selection
+ */
+class EventProjectSelectionForm extends React.PureComponent<Props,State> {
+  constructor(props: Props): void {
+    super(props);
+    const event: EventData = props.project;
+    this.state = {
+      selectedProjects: event.event_projects ? event.event_projects.map(ProjectAPIUtils.projectFromAPIData) : [],
+      formIsValid: false,
+      formFields: {
+        selectedProjects: [],
+      },
+    };
+    
+    this.form = form.setup();
+    props.readyForSubmit(true, this.onSubmit.bind(this));
+    this.addProjectToSelectedProjects = this.addProjectToSelectedProjects.bind(this);
+  
+    this.projectIdsField = React.createRef();
+  }
+  
+  componentDidMount() {
+    // Initial validation check
+    this.form.doValidation.bind(this)();
+  }
+
+  onSubmit(submitFunc: Function): void {
+    this.setState({}, submitFunc);
+  }
+
+  onValidationCheck(formIsValid: boolean): void {
+    this.props.readyForSubmit(true);
+
+    if(formIsValid !== this.state.formIsValid) {
+      this.setState({formIsValid});
+      this.props.readyForSubmit(true);
+    }
+  }
+
+  addProjectToSelectedProjects(project: ProjectData): void {
+    if (projectSelectionStoreSingleton.includes(project)) {
+      return;
+    }
+
+    projectSelectionStoreSingleton.push(project);
+    this.projectIdsField.current.value = JSON.stringify();
+    this.setState({
+      selectedProjects: [...projectSelectionStoreSingleton],
+    });
+  }
+
+  render(): React$Node {
+    // TODO: Move styles to css file
+    return (
+      <div className="EditProjectForm-root">
+        <DjangoCSRFToken/>
+
+        <div style={{ marginBottom: '20px' }}>
+          Which projects are participating in your event?
+        </div>
+  
+        <input type="hidden" ref={this.projectIdsField} id="" name=""/>
+        
+        <
+          div
+          className="SelectProjectsPillsContainer"
+          style={{
+            marginBottom: '20px'
+          }}
+        >
+          {
+            this.state.selectedProjects.map(project => {
+              // need pill buttons or tags
+              return (
+                <span
+                  style={{
+
+                    borderRadius: '20px',
+                    backgroundColor: '#F3A73C',
+                    height: '40px',
+                    padding: '10px 30px',
+                    display: 'flex',
+                    display: 'inline-block',
+                    margin: '0 10px 10px 0',
+
+                  }}
+                
+                >
+                  <div style={{
+                    display: 'flex',
+                    alignItems: 'center',
+                    justifyContent: 'center',
+                    color: '#fff',
```

Comment:
Just putting this TODO into a Github comment so we don't lose it. Move out all of these style rules to CSS classes.

Reply:
I did some preliminary work on this in 105d5cb but as the component appears to be commented out, I didn't progress further other than cleanup and moving style code into classes.